src/lib/operators/table_scan/between_table_scan_impl.cpp

```diff
@@ -0,0 +1,107 @@
+#include "between_table_scan_impl.hpp"
+
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include "storage/chunk.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/segment_iterables/create_iterable_from_attribute_vector.hpp"
+#include "storage/table.hpp"
+
+#include "utils/assert.hpp"
+
+#include "resolve_type.hpp"
+#include "type_comparison.hpp"
+
+namespace opossum {
+
+BetweenTableScanImpl::BetweenTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID left_column_id,
+                                           const AllTypeVariant& left_value, const AllTypeVariant& right_value)
+    : BaseSingleColumnTableScanImpl{in_table, left_column_id, PredicateCondition::Between},
+      _left_value{left_value},
+      _right_value{right_value} {}
+
+void BetweenTableScanImpl::handle_segment(const BaseValueSegment& base_segment,
+                                          std::shared_ptr<SegmentVisitorContext> base_context) {
+  auto context = std::static_pointer_cast<Context>(base_context);
```

Comment:
both for value- and encoded (non-dictionary) segments, you do a linear scan over the values. Why do we need to functions for this (and risk subtle bugs on encoded data). Wouldn't one function using `resolve_data_and_segment_type` be sufficient?

Reply:
That problem exists in SingleColumnTableScanImpl as well. Let me pull that into a separate issue.